pandas/io/feather_format.py

```diff
@@ -122,14 +127,15 @@ def read_feather(
     import_optional_dependency("pyarrow")
     from pyarrow import feather
 
-    path, _, _, should_close = get_filepath_or_buffer(
-        path, storage_options=storage_options
-    )
+    ioargs = get_filepath_or_buffer(path, storage_options=storage_options)
 
-    df = feather.read_feather(path, columns=columns, use_threads=bool(use_threads))
+    df = feather.read_feather(
+        ioargs.filepath_or_buffer, columns=columns, use_threads=bool(use_threads)
+    )
 
     # s3fs only validates the credentials when the file is closed.
-    if should_close:
-        path.close()
+    if ioargs.should_close:
+        assert isinstance(ioargs.filepath_or_buffer, (IO, IOBase))
```

Comment:
i guess mypy is picky here (e.g. have to do the asserts, but prob ok)

Reply:
I will probably change that to `assert not isinstance(..., str)` in case there is some weird cornercase were the buffer object might have a different type (would imply that the type annotations are wrong/incomplete but at least it wouldn't break at runtime).